app/queries/reasons_for_rejection_applications_query.rb

```diff
@@ -12,10 +14,10 @@ def call
       .where(current_recruitment_cycle_year: recruitment_cycle_year)
       .where.not(structured_rejection_reasons: nil)
       .order(created_at: :desc)
-      .page(filters[:page])
-      .per(30)
 
-    apply_filters(application_choices)
+    application_choices = apply_filters(application_choices)
+    @pagy, results = pagy(application_choices, page: filters[:page], items: 30)
+    results
```

Comment:
This is strange, I know it's basically copying the existing funcationality, but pagination should be in the query class. Also, we shouldn't be setting instance variables here either. This `@pagy` isn't being used at all, is it?